### DISCUSSION TITLE
Previews never really updating (Quarto + Python + Shiny + Jupyter)

### DISCUSSION THREAD

**Discussion by mmchugh87** [quarto-dev/quarto-cli#9753] [Q&A]
### Description

Hi there.

I'm on a windows machine. 
Python 3. 
Recent version of Shiny.
quarto-1.4.554-win.
I have installed the Jupyter-Quarto extension via jupyterlab extension manager.

I'm trying to develop my skills at building Shiny dashboards.

I have a very unremarkable Jupyter notebook, I've called it "penguin_jupyter.ipynb". 
It is basically a duplicate of the same content you will find in the Quarto penguin dashboard example.
Albeit, the opening lines in my case are these:

````
---
title: "Dirty Penguinz"
author: "Codswollop"
format: dashboard
server: shiny
execute: 
  enabled: true
---
````

I run the following line in a Terminal:
````
quarto preview penguin_jupyter.ipynb
````
The response in the terminal seems to be free from error. It looks like this:

````
Starting python3 kernel...Done

Executing 'penguin_jupyter.ipynb'
  Cell 1/6: ''...Done
  Cell 2/6: ''...Done
  Cell 3/6: ''...Done
  Cell 4/6: ''...Done
  Cell 5/6: ''...Done
  Cell 6/6: ''...Done

pandoc 
  to: html
  output-file: penguin_jupyter.html
  template: >-
    C:\Users\maste\AppData\Local\Programs\Quarto\share\formats\dashboard\template.html
  standalone: true
  section-divs: true
  html-math-method: mathjax
  wrap: none
  default-image-extension: png
  
metadata
  document-css: false
  link-citations: true
  date-format: long
  lang: en
  page-layout: custom
  title: Dirty Penguins
  author: Codswollop
  server:
    type: shiny
  remove-hidden: all
  
Output created: penguin_jupyter.html
````

If I make a trivial change to the jupyter file, and save it, I would hope the change would be reflected in the preview (which I am viewing on "localhost" using Chrome browser). However the changes are never reflected there. I have tried hitting refresh, and clearing the browser cache, but It doesn't help. 

The strange thing is, I can see that the output html (penguin_jupyter.html) is in fact adopting the changes I am making to the jupyter file... but this is not sufficient for the preview in the browser to be updated.

Can anyone assist? I have seen one discussion here that was only somewhat similar (#9202). I have also been looking at the Quarto Jupyter guide and it implies that my preview should be updating.

Many thanks

**Comment by mcanouil** [comment]
Could you properly format your post using code blocks for code and terminal outputs? Thanks.
If your code contains code blocks, you need to enclose it using more backticks, *i.e.*, usually four ` ```` `.

See <https://quarto.org/bug-reports.html>.

**Comment by mmchugh87** [comment]
Incidentally, I experience the same issue with Quarto Pre-release — v1.5.39.

What tends to happen is, to visualise any changes I have made in the Jupyter notebook, it seems I need to save the file with a new name and then open another terminal instance and preview the new file.

I end up with many, many terminal instances running simultaneously (I cannot terminate them no matter what I try).

If I try to preview the same file via two different terminal instances, I encounter errors (hence I have been changing the file names).

If I can get the preview feature working correctly, it will enormously increase the speed at which I can learn how Quarto / Shiny / Python work together.

**Comment by cderv** [comment]
We currently have a problem in our windows builds with preview
* https://github.com/quarto-dev/quarto-cli/issues/9574

This is probably related. 

We'll fix soon. Sorry for the trouble.

**Reply by mmchugh87** [reply]
thanks.
I have been playing around with quarto for the last 72 hours (including uninstalling, reinstalling, different versions).
I definitely encountered the missing "quarto-preview.js" error a day or two again but I cannot reproduce it now.
And I don't encounter that error anymore (but preview nonetheless does not work).

No idea if it is helpful to know but in my area here:
C:\Users\maste\AppData\Local\Programs\Quarto\share\preview
I have the file "quarto-preview" albeit it is a "chrome HTML document".
